sdk/clientcore/core/src/main/java/io/clientcore/core/implementation/instrumentation/fallback/FallbackContextPropagator.java

```diff
@@ -0,0 +1,116 @@
+
+    private FallbackContextPropagator() {
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <C> void inject(InstrumentationContext spanContext, C carrier, TraceContextSetter<C> setter) {
+        if (spanContext.isValid()) {
+            setter.set(carrier, "traceparent",
+                "00-" + spanContext.getTraceId() + "-" + spanContext.getSpanId() + "-" + spanContext.getTraceFlags());
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <C> InstrumentationContext extract(InstrumentationContext context, C carrier, TraceContextGetter<C> getter) {
+        String traceparent = getter.get(carrier, "traceparent");
+        if (traceparent != null) {
+            if (isValidTraceparent(traceparent)) {
+                String traceId = traceparent.substring(3, 35);
+                String spanId = traceparent.substring(36, 52);
+                String traceFlags = traceparent.substring(53, 55);
+                return new FallbackSpanContext(traceId, spanId, traceFlags, true, Span.noop());
+            } else {
+                LOGGER.atVerbose()
+                    .addKeyValue(HTTP_REQUEST_HEADER_TRACEPARENT_KEY, traceparent)
+                    .log("Invalid traceparent header");
+            }
+        }
+        return context == null ? FallbackSpanContext.INVALID : context;
+    }
+
+    /**
+     * Validates the traceparent header according to <a href="https://www.w3.org/TR/trace-context/#traceparent-header-field-values">W3C Trace Context</a>
+     *
+     * @param traceparent the traceparent header value
+     * @return true if the traceparent header is valid, false otherwise
+     */
+    private static boolean isValidTraceparent(String traceparent) {
+        if (traceparent == null || traceparent.length() != 55) {
+            return false;
+        }
+
+        // valid traceparent format: <version>-<trace-id>-<span-id>-<trace-flags>
+        // version - only 00 is supported
+        if (traceparent.charAt(0) != '0'
+            || traceparent.charAt(1) != '0'
+            || traceparent.charAt(2) != '-'
+            || traceparent.charAt(35) != '-'
+            || traceparent.charAt(52) != '-') {
+            return false;
+        }
+
+        // trace-id - 32 lower case hex characters, all 0 is invalid
+        boolean isAllZero = true;
+        for (int i = 3; i < 35; i++) {
+            char c = traceparent.charAt(i);
+            if (!((c >= '0' && c <= '9') || (c >= 'a' && c <= 'f'))) {
+                return false;
+            }
+            if (c != '0') {
+                isAllZero = false;
+            }
+        }
+        if (isAllZero) {
+            return false;
+        }
+
+        // span-id - 16 lower case hex characters, all 0 is invalid
+        isAllZero = true;
+        for (int i = 36; i < 52; i++) {
+            char c = traceparent.charAt(i);
+            if (!((c >= '0' && c <= '9') || (c >= 'a' && c <= 'f'))) {
```

Comment:
How do you feel about changing this check to

`c < '0' || c > 'f' || (c > '9' && c < 'a')`

Seems much cleaner and could be more performant as it reduces out some conditionals.